engine/src/main/java/org/terasology/rendering/nui/editor/NUIEditorScreen.java

```diff
@@ -168,31 +188,55 @@ public boolean onKeyEvent(NUIKeyEvent event) {
         return false;
     }
 
+    @Override
+    public void update(float delta) {
+        super.update(delta);
+
+        ResourceUrn systemUrn = nuiEditorSystem.getSelectedUrn();
+        if (systemUrn != null && systemUrn != selectedUrn) {
+            selectFile(systemUrn);
+            nuiEditorSystem.setSelectedUrn(null);
+        }
+    }
+
```

Comment:
Constantly checking if a value has changed doesn't look to me like a good design, but maybe I am overlooking something and you have a good reason for doing it that way. If that is the case please add a javadoc.

Without thinking about too much I would say that you could let the screen offer a setSelectedUrn and getSelectedUrn that gets called by the system when the change occurs and that calls selectFile. That way you woudn't have to constantly check in the update method if a value has changed.